src/main/kotlin/com/avail/descriptor/representation/AbstractDescriptor.kt

```diff
@@ -0,0 +1,4202 @@
+	 */
+	val debugObjectSlots: Array<Array<ObjectSlotsEnum>?>
+
+	/**
+	 * Used for quickly checking integer fields when
+	 * [AvailObjectRepresentation.shouldCheckSlots] is enabled.
+	 */
+	val debugIntegerSlots: Array<Array<IntegerSlotsEnum>?>
+
+	/**
+	 * Whether an [object][AvailObject] using this descriptor can have more than
+	 * the minimum number of object slots. Populated automatically by the
+	 * constructor, based on the presence of an underscore at the end of its
+	 * final [ObjectSlotsEnum] name.
+	 */
+	private val hasVariableObjectSlots: Boolean
+
+	/**
+	 * Whether an [object][AvailObject] using this descriptor can have more than
+	 * the minimum number of integer slots. Populated automatically by the
+	 * constructor, based on the presence of an underscore at the end of its
+	 * final [IntegerSlotsEnum] name.
+	 */
+	private val hasVariableIntegerSlots: Boolean
+
+	/**
+	 * The minimum number of object slots an [AvailObject] can have if it uses
+	 * this [descriptor][AbstractDescriptor]. Does not include indexed slots
+	 * possibly at the end. Populated automatically by the constructor.
+	 */
+	private val numberOfFixedObjectSlots: Int
+
+	/**
+	 * The minimum number of integer slots an [AvailObject] can have if it uses
+	 * this descriptor. Does not include indexed slots possibly at the end.
+	 * Populated automatically by the constructor.
+	 */
+	private val numberOfFixedIntegerSlots: Int
+
+	init
+	{
+		val objectSlots: Array<out ObjectSlotsEnum> =
+			if (objectSlotsEnumClass != null) objectSlotsEnumClass.enumConstants
+			else arrayOf()
+		@Suppress("ConstantConditionIf")
+		debugObjectSlots =
+			if (AvailObjectRepresentation.shouldCheckSlots)
+				arrayOfNulls(max(objectSlots.size, 1))
+			else
+				emptyDebugObjectSlots
+		hasVariableObjectSlots =
+			objectSlots.isNotEmpty()
+				&& objectSlots[objectSlots.size - 1].fieldName().endsWith("_")
+		numberOfFixedObjectSlots =
+			objectSlots.size - if (hasVariableObjectSlots) 1 else 0
+		val integerSlots: Array<out IntegerSlotsEnum> =
+			if (integerSlotsEnumClass != null)
+				integerSlotsEnumClass.enumConstants
+			else
+				arrayOf()
+		@Suppress("ConstantConditionIf")
+		debugIntegerSlots =
+			if (AvailObjectRepresentation.shouldCheckSlots)
+				arrayOfNulls(max(integerSlots.size, 1))
+			else
+				emptyDebugIntegerSlots
+		hasVariableIntegerSlots =
+			integerSlots.isNotEmpty()
+				&& integerSlots[integerSlots.size - 1].fieldName().endsWith("_")
+		numberOfFixedIntegerSlots =
+			integerSlots.size - if (hasVariableIntegerSlots()) 1 else 0
+	}
+
+	/**
+	 * A non-enum [ObjectSlotsEnum] that can be instantiated at will.
+	 * Useful for customizing debugger views of objects.
+	 *
+	 * @property
+	 *   The slot name.
+	 *
+	 * @constructor
+	 *
+	 * Create a new artificial slot with the given name.
+	 *
+	 * @param name
+	 *   The name of the slot.
+	 */
+	class DebuggerObjectSlots (val name: String) : ObjectSlotsEnumJava
```

Comment:
If possible, use ObjectSlotsEnum.  ObjectSlotsEnumJava is temporary.

Reply:
It was not possible.